WPMedia/arc-themes-components
# TMEDIA-852 : Create Picture Component

## Ticket

- [TMEDIA-852 : TMEDIA-852 : Create Picture Component](https://arcpublishing.atlassian.net/browse/TMEDIA-852)

## Description

A `Picture` component and `Picture.Source` subcomponent was added to render `<picture>` and `<source>` tags.

## Acceptance Criteria

- Use `npm run generate:component` to scaffold out “Picture” component in the arc-themes-components repo
     - Type “picture” into the response for the component name 
- Replace placeholder jsx with a `<picture>` element 
     - Picture element should set component class name and any additional classnames. 
- Create storybook stories using a fallback <img> element within the <picture> element, using  <Image>'s component
```
<Picture 
  // other picture sub components
  <Image 
    src="my-image"
    alt="Description of the image"
    height={1600}
    width={900}
    resizerOptions={{
      auth: "abc123",
      filter: 70,
      smart: true,
    }}
  />
/>
```
```
<picture>
  
  <img
    src="my-image?auth=abc123&width=1600&height=900&filter=70&smart=true"
    width="1600"
    height="900"
    
  >
</picture>
```
- Create `Picture.Source` sub combonent. The sub component will be setup like the carousel
     - The source should follow a pattern similar to the carousel component where the source element is a sub component. (See carousel reference of [child](https://github.com/WPMedia/arc-themes-components/blob/arc-themes-release-version-2.00/src/components/carousel/_children/Item.jsx), [mapping](https://github.com/WPMedia/arc-themes-components/blob/arc-themes-release-version-2.00/src/components/carousel/index.jsx#L409), and[ parent example](https://61eeeb27215b5f003acdf18f-otyyhrauqw.chromatic.com/?path=/docs/components-carousel--carousel).)
- Takes in an object resizerOptions. Note: A source image token is only compatible with one source image
```
<Picture>
  <Picture.Source 
    src="my-image-portrait"
    media="(min-width: 800px)"
    height={1600}
    width={900}
    resizerOptions={
      auth: "xyz123",
      filter: 70,
      smart: true,
    }
  />
   <Picture.Source 
    src="my-image-skinnier-screen"
    media="(max-width: 799px)"
    height={50}
    width={100}
    resizerOptions={{
      auth: "fff1111",
      filter: 70,
      smart: true,
    }}
  />
  
  <Image
    src="my-image"
    height={1600}
    width={900}
    resizerOptions={{
      auth: "abc123",
      filter: 70,
      smart: true,
    }}
    alt="Description of the Image"
  />
</Picture>
```
```
<picture>
  <source 
    srcset="my-image-portrait?auth=xyz123&width=900&height=1600&filter=70&smart=true" 
    media="(min-width: 800px)"
    width="1600"
    height="900"
  >
  <source 
    srcset="my-image-skinnier-screen?auth=fff1111&width=100&height=50&filter=70&smart=true" 
    media="(max-width: 799px)"
    width="100"
    height="50"
  >
  <img 
    src="my-image?auth=abc123&width=1600&height=900&filter=70&smart=true"
    width="1600"
    height="900"
    alt="Description of the Image"
  >
</picture>
```
- If no media query is available from `mediaCondition`, do not render the source tag
- If no src, do not render the `source` tag 
- It’s ok if no height or width provided. Do not render those properties if missing.
- Storybook examples should show various options for media
     - media queries min and max width
     - Not rendering the source tag if no `mediaCondition`
- Image source mapping functions should be handled in the utils folder
- Image should not be able to be stretched past its intrinsic size
- Test coverage should be 100% 
- Documentation should be written in the style of other components, including usage and proptypes. The audience should be for developers as well as some basic information for non-technical users. 
     - Include links to the resizer docs https://github.com/WPMedia/arc-photo-resizer/tree/main/docs/usage#query-parameters
-  Note: “types” file like “webp” are not necessary using v2 resizer

## Test Steps

1. Checkout branch - `git checkout {TMEDIA-852-create-picture-component}`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the new storybook documentation

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._


## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
